web-scraper/src/pullArticles.py

```diff
@@ -0,0 +1,115 @@
+from bs4 import BeautifulSoup
+import json
+import numpy as np
+import requests
+from requests.models import MissingSchema
+import spacy
+import trafilatura
+import lxml
+from constant import apiKey
+
+def beautifulsoup_extract_text_fallback(response_content):
+    
+    '''
+    This is a fallback function, so that we can always return a value for text content.
+    Even for when both Trafilatura and BeautifulSoup are unable to extract the text from a 
+    single URL.
+    '''
+    
+    # Create the beautifulsoup object:
+    soup = BeautifulSoup(response_content, 'html.parser')
+    
+    # Finding the text:
+    text = soup.find_all(string=True)
+    
+    # Remove unwanted tag elements:
+    cleaned_text = ''
+    blacklist = [
+        '[document]',
+        'noscript',
+        'header',
+        'meta',
+        'html',
+        'head', 
+        'input',
+        'script',
+        'style',
+        'nav',
+        'footer']
+
+    # Then we will loop over every item in the extract text and make sure that the beautifulsoup4 tag
+    # is NOT in the blacklist
+    for item in text:
+        if item.parent.name not in blacklist:
+            cleaned_text += '{} '.format(item)
+            
+    # Remove any tab separation and strip the text:
+    cleaned_text = cleaned_text.replace('\t', '')
+    
+    
+    return cleaned_text.strip()
+
+
+
+
+headers = {'Accept': 'application/json'}
+
+articleSources = requests.get('https://newsapi.org/v2/everything?q=samsungs&from=2024-11-15&sortBy=publishedAt&apiKey='+apiKey)
```

Comment:
@FlorianMimmler @MartinOkoli eventuell auslagern in eien funktion mit company name als parameter und datum